### PR TITLE
BUG: RollingGroupby no longer keeps the groupby column in the result

### LINKED ISSUE #32262: Rolling groupby should not maintain the by column in the resulting DataFrame
When performing a `rolling` operation on a `groupby` object, the index level will be incorrectly maintained in the dataframe.

```
import numpy as np
import pandas as pd

df = pd.DataFrame({'idx_L1': ['Level_1'] * 10 + ['Level_2'] * 10 + ['Level_3'] * 10,  
                   'idx_L2': ([1] * 5 + [2] * 5 + [3] * 5) * 2,  
                   'vals': np.arange(30)})

df.groupby(['idx_L1', 'idx_L2']).rolling(3).sum()
```

This issue has been discussed in #14013.

It has been closed eventhough the bug still exists in v1.01.

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #32262
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

Borrowed some work from https://github.com/pandas-dev/pandas/pull/32332

### CODE REVIEW

**Comment by jreback** [comment]
cc @pandas-dev/pandas-core, @pandas-dev/pandas-triage  if any comments (this was a discussion item on the call i think month before last)

**Comment by mroeschke** [comment]
https://github.com/pandas-dev/pandas/issues/38787 was the discussion item I believe you are referring to @jreback which this PR does not address

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -558,6 +558,10 @@ def __init__(
         if _grouper is None:
             raise ValueError("Must pass a Grouper object.")
         self._grouper = _grouper
+        # GH 32262: It's convention to keep the grouping column in
```
Comment (jbrockmendel) [inline pandas/core/window/rolling.py:561]: is the difference in conventions just a historical thing?  might we want to make these match eventually?

**Review by jbrockmendel — APPROVED** [review]
LGTM

File: pandas/core/window/rolling.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/window/rolling.py:561]: Seems to be historical https://github.com/pandas-dev/pandas/pull/32332#pullrequestreview-367673502

Would be great to have groupby apply match but might take come convincing

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**